pkg/main.go

```diff
@@ -19,25 +20,38 @@ func main() {
 	msgCh2 := make(chan messages.Message, bufLen)
 
 	bufLen = GetBufferLength("ACK_BUFFER_LENGTH")
-	ackCh := make(chan messages.Ack, bufLen)
+	ackCh1 := make(chan messages.Message, bufLen)
+	ackCh2 := make(chan messages.Message, bufLen)
 
-	reader, err := readers.NewReader(msgCh1, ackCh)
+	reader, err := readers.NewReader(msgCh1, ackCh1)
```

Comment:
maybe instead of 1 and 2, a better name could be used. The same for msgCh.

Reply:
Have you though about something?

At the end msgCh1 and msgCh2 is the same channel but split in two parts in order to allow middleware in the middle (the same is valid for ackCh1 and ackCh2).

What about: `readerInputCh`, `readerOutputCh`, `writerInputCh`, `writerOuputCh`?